### PR TITLE
feat: add test delete user

### PR DESCRIPTION
**Pull request by Anton-Hudz**

### CODE REVIEW

File: app/transport/rest/user_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+			uh := UserHandler{
+				usecase: tt.usecaseConstructor(ctrl),
+				logger:  logger,
+			}
+
+			resp := httptest.NewRecorder()
+			uh.DeleteUser(resp, httptest.NewRequest(http.MethodDelete, "*", strings.NewReader(tt.requestBody)))
+			require.Equal(t, tt.expStatusCode, resp.Code)
+			if name != "succes" {
```
Comment (vpakhuchyi) [inline app/transport/rest/user_test.go:126]: That's not good.
We should not have specific conditions here.
If you don't need to check body in this test case - try to describe that in a test case itself.

File: app/transport/rest/user_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+				mock := NewMockUserUsecase(ctrl)
+
+				mock.EXPECT().Delete(entities.User{
+					ID: "dca5947d-3dfc-49f1-bc09-dd53ce7e71cc",
+				}).Return(nil).Times(1)
+
+				return mock
+			},
+			expResponseBody: "",
+			expStatusCode:   204,
```
Comment (vdshakh) [inline app/transport/rest/user_test.go:41]: There is an idea that it is better to replace such magic numbers with status codes from the http package

Comment (Anton-Hudz) [reply app/transport/rest/user_test.go:41]: Perhaps, it is good idea

**Review by Rusli4k — DISMISSED** [review]

File: app/transport/rest/user_test.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+				mock.EXPECT().Delete(entities.User{
+					ID: "dca5947d-3dfc-49f1-bc09-dd53ce7e71cc",
+				}).Return(nil).Times(1)
+
+				return mock
+			},
+			expResponseBody: "",
+			expStatusCode:   http.StatusNoContent,
+		},
+		"invalid_undefined_id": {
```
Comment (Brigant) [inline app/transport/rest/user_test.go:43]: May be `notexisted-id` is better name for this test case?

Comment (Anton-Hudz) [reply app/transport/rest/user_test.go]: Done

File: app/transport/rest/user_test.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+	"github.com/go-devs-ua/octagon/app/entities"
+	"github.com/go-devs-ua/octagon/app/globals"
+	"github.com/go-devs-ua/octagon/lgr"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUserHandler_DeleteUser(t *testing.T) {
+	logger, err := lgr.New("INFO")
```
Comment (vdshakh) [inline app/transport/rest/user_test.go:18]: Maybe it's better to replace "INFO" with a constant from the lgr package?

Comment (Anton-Hudz) [reply app/transport/rest/user_test.go:21]: Ok, done

**Review by vdshakh — DISMISSED** [review]

**Review by Brigant — DISMISSED** [review]

File: app/transport/rest/user_test.go [hunk 5]
```diff
@@ -294,3 +296,100 @@ func TestUserHandler_GetUserByID(t *testing.T) {
...
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+
+			uh := UserHandler{
+				usecase: tt.usecaseConstructor(ctrl),
+				logger:  logger,
+			}
+			defer ctrl.Finish()
```
Comment (vpakhuchyi) [inline app/transport/rest/user_test.go:386]: There is no sense to run it in defer.
Just call it after uh.DeleteUser() call.

Please recall how defer works in a loop.

Comment (Anton-Hudz) [reply app/transport/rest/user_test.go]: Ok, done

**Review by vpakhuchyi — APPROVED** [review]

**Review by Brigant — APPROVED** [review]

**Review by Rusli4k — APPROVED** [review]

**Review by vdshakh — APPROVED** [review]

**Review by Rusli4k — APPROVED** [review]

**Review by jiztastamablastamarang — APPROVED** [review]

**Review by jiztastamablastamarang — APPROVED** [review]

**Pull request merged**